tensorflow/lite/micro/examples/mnist_lstm/evaluate.py

```diff
@@ -0,0 +1,107 @@
+# Copyright 2022 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =============================================================================
+"""
+LSTM model evaluation for MNIST recognition
+
+Run:
+bazel build tensorflow/lite/micro/examples/mnist_lstm:evaluate
+bazel-bin/tensorflow/lite/micro/examples/mnist_lstm/evaluate --model_path='.tflite file path' --img_path='MNIST image path'
+
+"""
+import os
+
+import numpy as np
+from PIL import Image
+
+from absl import app
+from absl import flags
+from absl import logging
+
+from tflite_micro.tensorflow.lite.micro.python.interpreter.src import tflm_runtime
+
+FLAGS = flags.FLAGS
+
+flags.DEFINE_string('model_path', '/tmp/lstm_trained_model/lstm.tflite',
+                    'the trained model path.')
+flags.DEFINE_string('img_path', "/tmp/samples/sample0.jpg",
+                    'path for the image to be predicted.')
+
+
+def read_img(img_path):
+  """Read MNIST image
+
+  Args:
+      img_path (str): path to a MNIST image
+
+  Returns:
+      np.array : image in the correct np.array format
+  """
+  image = Image.open(img_path)
+  data = np.asarray(image, dtype=np.float32)
+  data = data / 255.0
+  data = data.reshape((1, 28, 28))  #batch size one
```

Comment:
perhaps add some checks that the image size is what the model expects? and add a test that passes in an image that is not of the desired size and confirm that the exception is raised.

http://go/codelab-test-python#testing-exceptional-conditions

Reply:
Done.